frontend/src/routes/protected.route.tsx

```diff
@@ -0,0 +1,21 @@
+import { useAppSelector } from "@/store/hooks";
+import React from "react";
+import { Navigate, Route, Routes, useLocation } from "react-router-dom";
+
+type RouteProps = {
+  path: string;
+  children: React.ReactNode;
+};
+const ProtectedRoute = ({ path, children }: RouteProps) => {
+  const { isLoggedIn } = useAppSelector((state) => state.auth);
+  let location = useLocation();
+  return isLoggedIn ? (
+    <Routes>
+      <Route path={path}>{children}</Route>
+    </Routes>
+  ) : (
+    <Navigate to={"/"} state={{ from: location }} replace={true} />
+  );
+};
```

Comment:
**suggestion :** we can simplify the code like below

```typescript

const ProtectedRoute = ({ path, children }: RouteProps) => {
  const { isLoggedIn } = useAppSelector((state) => state.auth);
  const navigate = useNavigate();

  if(!isLoggedIn){
   return <Navigate to={ROUTES.LOGIN} replace={true} />
  }

  return children;
};

```

please check if above code works 😎